be/src/cloud/pb_convert.cpp

```diff
@@ -134,19 +137,55 @@ void doris_rowset_meta_to_cloud(RowsetMetaCloudPB* out, RowsetMetaPB&& in) {
     out->set_has_variant_type_in_schema(in.has_variant_type_in_schema());
 }
 
-RowsetMetaPB cloud_rowset_meta_to_doris(const RowsetMetaCloudPB& in) {
+RowsetMetaPB cloud_rowset_meta_to_doris(const RowsetMetaCloudPB& in,
+                                        std::optional<SchemaCloudDictionary> dict) {
     RowsetMetaPB out;
-    cloud_rowset_meta_to_doris(&out, in);
+    cloud_rowset_meta_to_doris(&out, in, dict);
     return out;
 }
 
-RowsetMetaPB cloud_rowset_meta_to_doris(RowsetMetaCloudPB&& in) {
+RowsetMetaPB cloud_rowset_meta_to_doris(RowsetMetaCloudPB&& in,
+                                        std::optional<SchemaCloudDictionary> dict) {
     RowsetMetaPB out;
-    cloud_rowset_meta_to_doris(&out, std::move(in));
+    cloud_rowset_meta_to_doris(&out, std::move(in), dict);
     return out;
 }
 
-void cloud_rowset_meta_to_doris(RowsetMetaPB* out, const RowsetMetaCloudPB& in) {
+static void fill_schema_with_dict(const RowsetMetaCloudPB& in, RowsetMetaPB* out,
```

Comment:
Move it into `MetaServiceImpl::get_rowset` so BE doesn't need to know the `SchemaCloudDictionary `.

Reply:
@gavinchou  required, pls take a look

Reply:
done